### PR TITLE
Simplify tests setup

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- Only set image name in job wrapper
- Obtain scheme from client when building a reconciler
- Testing util to build a context with verbose logging

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
Created #951 for flaky test

/retest

File: pkg/util/testing/context.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/testr"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func ContextWithLog(t *testing.T) (context.Context, logr.Logger) {
```
Comment (trasc) [inline pkg/util/testing/context.go:28]: can we add a parameter for `Verbosity`? and maybe add a value to the context to potentially identify the test?

**Review by trasc — COMMENTED** [review]
nits

File: pkg/util/testing/context.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/util/testing/context.go:28]: I think verbosity should come from a command line flag, but we can leave that for a follow up.

The testr logger already accepts `t`. Then any logs lines are already contextualized for the test. Furthermore, they are only showed if the test fails.

**Pull request merged**